docs/topics/testing/tools.txt

```diff
@@ -153,23 +158,29 @@ Use the ``django.test.Client`` class to make requests.
 
             /customers/details/?name=fred&age=7
 
-        The ``extra`` keyword arguments parameter can be used to specify
-        headers to be sent in the request. For example::
+        The ``headers`` parameter and ``extra`` keyword arguments parameter can
+        be used to specify headers to be sent in the request. For example::
 
             >>> c = Client()
             >>> c.get('/customers/details/', {'name': 'fred', 'age': 7},
             ...       HTTP_ACCEPT='application/json')
 
+        or
+
+            >>> c = Client()
+            >>> c.get('/customers/details/', {'name': 'fred', 'age': 7},
+            ...       headers={'Accept': 'application/json'})
```

Comment:
document `headers` first as the canonical way to pass headers

Then something like “Arbitrary keyword arguments set WSGI [environ variables](https://peps.python.org/pep-3333/#environ-variables). For example, headers are set with `HTTP_*` syntax: [example]”. There will still be rare use cases for setting other environ variables.

Also check against ASGI spec...